pvlib/test/test_pvsystem.py

```diff
@@ -82,162 +79,39 @@ def test_systemdef_dict():
     assert expected == pvsystem.systemdef(meta, 5, 0, .1, 5, 5)
 
 
-@needs_numpy_1_10
-def test_ashraeiam():
-    thetas = np.array([-90. , -67.5, -45. , -22.5,   0. ,  22.5,  45. ,  67.5, 89.,  90. , np.nan])
-    iam = pvsystem.ashraeiam(thetas, .05)
-    expected = np.array([        0,  0.9193437 ,  0.97928932,  0.99588039,  1.        ,
-        0.99588039,  0.97928932,  0.9193437 ,         0, 0,  np.nan])
-    assert_allclose(iam, expected, equal_nan=True)
-
-
-@needs_numpy_1_10
-def test_ashraeiam_scalar():
-    thetas = -45.
-    iam = pvsystem.ashraeiam(thetas, .05)
-    expected = 0.97928932
-    assert_allclose(iam, expected, equal_nan=True)
-    thetas = np.nan
-    iam = pvsystem.ashraeiam(thetas, .05)
-    expected = np.nan
-    assert_allclose(iam, expected, equal_nan=True)
-
-
-def test_PVSystem_ashraeiam(mocker):
-    mocker.spy(pvsystem, 'ashraeiam')
-    module_parameters = pd.Series({'b': 0.05})
-    system = pvsystem.PVSystem(module_parameters=module_parameters)
-    thetas = 1
-    iam = system.ashraeiam(thetas)
-    pvsystem.ashraeiam.assert_called_once_with(thetas, b=0.05)
-    assert iam < 1.
-
-
-@needs_numpy_1_10
-def test_physicaliam():
-    aoi = np.array([-90. , -67.5, -45. , -22.5,   0. ,  22.5,  45. ,  67.5,  90. , np.nan])
-    iam = pvsystem.physicaliam(aoi, 1.526, 0.002, 4)
-    expected = np.array([        0,  0.8893998,  0.98797788,  0.99926198,         1,
-        0.99926198,  0.98797788,  0.8893998,         0, np.nan])
-    assert_allclose(iam, expected, equal_nan=True)
-
-    # GitHub issue 397
-    aoi = pd.Series(aoi)
-    iam = pvsystem.physicaliam(aoi, 1.526, 0.002, 4)
-    expected = pd.Series(expected)
-    assert_series_equal(iam, expected)
-
-
-@needs_numpy_1_10
-def test_physicaliam_scalar():
-    aoi = -45.
-    iam = pvsystem.physicaliam(aoi, 1.526, 0.002, 4)
-    expected = 0.98797788
-    assert_allclose(iam, expected, equal_nan=True)
-    aoi = np.nan
-    iam = pvsystem.physicaliam(aoi, 1.526, 0.002, 4)
-    expected = np.nan
-    assert_allclose(iam, expected, equal_nan=True)
-
-
-def test_PVSystem_physicaliam(mocker):
-    module_parameters = pd.Series({'K': 4, 'L': 0.002, 'n': 1.526})
-    system = pvsystem.PVSystem(module_parameters=module_parameters)
-    mocker.spy(pvsystem, 'physicaliam')
+@pytest.mark.parametrize('iam_model,model_params', [
+    ('ashrae', {'b': 0.05}),
+    ('physical', {'K': 4, 'L': 0.002, 'n': 1.526}),
+    ('martin_ruiz', {'a_r': 0.16}),
+])
+def test_PVSystem_get_iam(mocker, iam_model, model_params):
+    m = mocker.spy(_iam, iam_model)
+    system = pvsystem.PVSystem(module_parameters=model_params)
     thetas = 1
-    iam = system.physicaliam(thetas)
-    pvsystem.physicaliam.assert_called_once_with(thetas, **module_parameters)
+    iam = system.get_iam(thetas, iam_model=iam_model)
+    assert m.assert_called_once_with(thetas, **model_params)
```

Comment:
The problem is the extra `assert` at the beginning of the line. Python first calls `m.assert_called_once_with(thetas, **model_params)`, which returns `None` if it does not error. Then the line looks like `assert None`. which always fails.